Lib/zipfile.py

```diff
@@ -1366,7 +1366,10 @@ def _RealGetContents(self):
             print("given, inferred, offset", offset_cd, inferred, concat)
         # self.start_dir:  Position of start of central directory
         self.start_dir = offset_cd + concat
-        fp.seek(self.start_dir, 0)
+        try:
+            fp.seek(self.start_dir, 0)
+        except ValueError as e:
+            raise BadZipFile("Bad offset for central directory")  from e
```

Comment:
"from e" is not needed here, because by default the new raised exception is linked to the handled exception.

It is even better to suppress this, because this ValueError is an implementation detail. The code could be written as `if self.start_dir < 0: raise ...`, and there would not be any ValueError.

```suggestion
        except ValueError:
            raise BadZipFile("Bad offset for central directory")
```

Reply:
Resolved in https://github.com/python/cpython/pull/32291/commits/bf8d1670c9726301c77beb00323d62eaa478a8a6